enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,355 @@
+
+Since WMCO is a day 2 operator, it will pick up mirror settings during runtime regardless of if they were set during
+cluster install time or at some point during the cluster's lifetime. When mirror settings are updated, WMCO will react by:
+- creating a new set of hosts.toml files
+- copying over the new registry config files to each istance, if a file has changed
+
+All changes detailed in this enhancement proposal will be limited to the Windows Machine Config Operator and its 
+sub-component, Windows Instance Config Daemon (WICD).
+
+Example: A cluster contains the following resources
+```yaml
+apiVersion: config.openshift.io/v1 
+kind: ImageDigestMirrorSet 
+metadata:
+  name: ubi9repo
+spec:
+  imageDigestMirrors: 
+  - mirrors:
+    - example.io/example/ubi-minimal
+    - example.com/example2/ubi-minimal
+    source: registry.access.redhat.com/ubi9/ubi-minimal 
+    mirrorSourcePolicy: AllowContactingSource 
+  - mirrors:
+    - mirror.example.com
+    source: registry.redhat.io
+    mirrorSourcePolicy: NeverContactSource
+```
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageTagMirrorSet
+metadata:
+  name: tag-mirror
+spec:
+  imageTagMirrors:
+  - mirrors:
+    - docker.io
+    source: docker-mirror.internal
+    mirrorSourcePolicy: AllowContactingSource
+```
+
+would result in WMCO creating 3 files to place on each Windows instance.
+```powershell
+$ tree $config_path
+C:/k/containerd/registries/
+|── registry.access.redhat.com
+|   └── hosts.toml
+|── mirror.example.com
+|   └── hosts.toml
+└── docker.io
+    └── hosts.toml
+
+
+$ cat "$config_path"/registry.access.redhat.com/host.toml
+server = "https://registry.access.redhat.com" # default fallback server since "AllowContactingSource" mirrorSourcePolicy is set
+
+[host."https://example.io/example/ubi-minimal"]
+ capabilities = ["pull"]
+
+[host."https://example.com/example2/ubi-minimal"] # secondary mirror
+ capabilities = ["pull"]
+
+
+$ cat "$config_path"/registry.redhat.io/host.toml
+# "server" omitted since "NeverContactSource" mirrorSourcePolicy is set
+
+[host."https://mirror.example.com"]
+ capabilities = ["pull"]
+
+
+$ cat "$config_path"/docker.io/host.toml
+server = "https://docker.io"
+
+[host."https://docker-mirror.internal"]
+ capabilities = ["pull", "resolve"] # resolve tags
+```
+
+As on the Linux side, it is upon the user to ensure there is no clash between mirrors across their IDMS and ITMS resources.
+
+---
+
+### Test Plan & Infrastructure Needed
+
+In addition to unit testing individual WMCO/WICD packages and controllers, an e2e job will be added to the release repo
+for WMCO's master/release-4.15 branches. A new CI workflow will be created leveraging 
+[existing step-registry steps](https://github.com/openshift/release/tree/master/ci-operator/step-registry/aws/provision/vpc/disconnected),
+which creates a disconnected cluster in AWS or vSphere with hybrid-overlay networking. 
```

Comment:
> hybrid-overlay networking

is a WMCO requirement, not sure about the value added by mentioning it here.

Reply:
value here is that the existing workflows in the step registry don't bring up clusters with OVN hybrid out of the box. We'll have to add this step into our job's steps